src/services/dealer-price/index.ts

```diff
@@ -0,0 +1,179 @@
+import util from "util"
+
+import { credentials } from "@grpc/grpc-js"
+
+import { UnknownDealerPriceServiceError } from "@domain/dealer-price"
+
+import { toSats } from "@domain/bitcoin"
+
+import { baseLogger } from "../logger"
+
+import { PriceServiceClient } from "./proto/services/price/v1/price_service_grpc_pb"
+import {
+  GetExchangeRateForImmediateUsdBuyRequest,
+  GetExchangeRateForImmediateUsdBuyResponse,
+  GetExchangeRateForImmediateUsdBuyFromCentsRequest,
+  GetExchangeRateForImmediateUsdBuyFromCentsResponse,
+  GetExchangeRateForImmediateUsdSellRequest,
+  GetExchangeRateForImmediateUsdSellResponse,
+  GetExchangeRateForImmediateUsdSellFromSatoshisRequest,
+  GetExchangeRateForImmediateUsdSellFromSatoshisResponse,
+  GetQuoteRateForFutureUsdBuyRequest,
+  GetQuoteRateForFutureUsdBuyResponse,
+  GetQuoteRateForFutureUsdSellRequest,
+  GetQuoteRateForFutureUsdSellResponse,
+} from "./proto/services/price/v1/price_service_pb"
+
+const serverPort = process.env.PRICE_SERVER_PORT ?? "50055"
+const client = new PriceServiceClient(
+  `localhost:${serverPort}`,
+  credentials.createInsecure(),
+)
+
+const clientGetExchangeRateForImmediateUsdBuy = util.promisify<
+  GetExchangeRateForImmediateUsdBuyRequest,
+  GetExchangeRateForImmediateUsdBuyResponse
+>(client.getExchangeRateForImmediateUsdBuy.bind(client))
+
+const clientGetExchangeRateForImmediateUsdBuyFromCents = util.promisify<
+  GetExchangeRateForImmediateUsdBuyFromCentsRequest,
+  GetExchangeRateForImmediateUsdBuyFromCentsResponse
+>(client.getExchangeRateForImmediateUsdBuyFromCents.bind(client))
+
+const clientGetExchangeRateForImmediateUsdSell = util.promisify<
+  GetExchangeRateForImmediateUsdSellRequest,
+  GetExchangeRateForImmediateUsdSellResponse
+>(client.getExchangeRateForImmediateUsdSell.bind(client))
+
+const clientGetExchangeRateForImmediateUsdSellFromSatoshis = util.promisify<
+  GetExchangeRateForImmediateUsdSellFromSatoshisRequest,
+  GetExchangeRateForImmediateUsdSellFromSatoshisResponse
+>(client.getExchangeRateForImmediateUsdSellFromSatoshis.bind(client))
+
+const clientGetQuoteRateForFutureUsdBuy = util.promisify<
+  GetQuoteRateForFutureUsdBuyRequest,
+  GetQuoteRateForFutureUsdBuyResponse
+>(client.getQuoteRateForFutureUsdBuy.bind(client))
+
+const clientGetQuoteRateForFutureUsdSell = util.promisify<
+  GetQuoteRateForFutureUsdSellRequest,
+  GetQuoteRateForFutureUsdSellResponse
+>(client.getQuoteRateForFutureUsdSell.bind(client))
+
+export const DealerPriceService = (): IDealerPriceService => {
+  const getExchangeRateForImmediateUsdBuy = async function (
+    amountInSatoshis: Satoshis,
+  ): Promise<UsdCents | DealerPriceServiceError> {
+    try {
+      const response = await clientGetExchangeRateForImmediateUsdBuy(
+        new GetExchangeRateForImmediateUsdBuyRequest().setAmountInSatoshis(
+          amountInSatoshis,
+        ),
+      )
+      return response.getAmountInUsd() as UsdCents
```

Comment:
I just noticed from another PR that we have a `toCents` method now as well (similar to `toSats`). Sorry I missed this before 🙈

Reply:
Yeah no worries I had to do a second pass to convert all to Cents on that interface anyways so very timely! Thx